src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQueryMultiSink.java

```diff
@@ -50,7 +51,7 @@
   + "Data is first written to a temporary location on Google Cloud Storage, then loaded into BigQuery from there.")
 public class BigQueryMultiSink extends AbstractBigQuerySink {
   private static final String TABLE_PREFIX = "multisink.";
-
+  private static final String OUTPUT_PATTERN = "[A-Za-z0-9_-]+";
```

Comment:
From [this regex](https://github.com/cdapio/cdap/blob/2de0ef365a77ef549d071e52c89e73a0b0652b93/cdap-proto/src/main/java/io/cdap/cdap/proto/id/EntityId.java#L67), it seems like  '.' and '$' for dataset ids.
Can you please try to reproduce the issue and figure out the exact cause of the failure?

Reply:
First of all, are we talking about CDAP Datasets?

Because according to the description in JIRA:
`The BigQuery multi sink fails when tables have characters that aren't allowed by CDAP's dataset system (ex: $).`

We used this documentation as reference for CDAP Dataset Names: https://cdap.atlassian.net/wiki/spaces/DOCS/pages/477593783/Naming+Conventions

In here, the output name is generated with combination of table name, but the table name is validated before creating outputName, so if there is any issue will validation, the pipeline with fail before creating the outputName. 

Please check this line of code: https://github.com/data-integrations/google-cloud/blob/799447e7ef8b8ff83f72fa149f18e06e0381a804/src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQueryMultiSink.java#L115

The only difference between CDAP Dataset system and BigQuery naming convention is that BQ table name can contain $ (in case when table is partition by time, for example **client$2020**).

When BQ table exists and is partitioned by time (ex. partition by year), then BQMultisink can write data to this specific table **client$2020**

When we tested with scenario, the pipeline was able to run successfully, but we were unable to view the Lineage Records in CDAP because of an unexpected error that says: 

`Document.querySelector: '#target_ns-PLUGIN_191_ds-mlt-client$2020_fd-surname' is not a valid selector` 

I think this was the reason why special characters were asked to be sanitized. 

Please correct me if my assumption is wrong.

Reply:
I see, thanks for the explanation!